File: Conceitos de LPs/Tipos de Dados/Ordinais/ordinais.cs
//C Sharp (C#)
//C# oferece suporte a enumerações, mas não a subfaixas.
using System;

class Ordinais {
    enum DiaDaSemana { Segunda, Terca, Quarta, Quinta, Sexta, Sabado, Domingo }

    static void Main() {
        DiaDaSemana hoje = DiaDaSemana.Terca;

        // Exibindo valor ordinal
        Console.WriteLine("Hoje é: " + hoje);

        // Usando a posição ordinal
        Console.WriteLine("Posição ordinal: " + (int)hoje);

        if (hoje == DiaDaSemana.Terca) {
            Console.WriteLine("Hoje é terça-feira.");
        }
    }
}

File: Conceitos de LPs/Tipos de Dados/Ordinais/ordinais.py
#Python
#Em Python, podemos usar o módulo Enum para criar enumerações, mas não há suporte nativo para subfaixas.
from enum import Enum

class DiaDaSemana(Enum):
    Segunda = 1
    Terca = 2
    Quarta = 3
    Quinta = 4
    Sexta = 5
    Sabado = 6
    Domingo = 7

hoje = DiaDaSemana.Terca

# Exibindo valor ordinal
print(f"Hoje é: {hoje.name}")

# Usando a posição ordinal
print(f"Posição ordinal: {hoje.value}")

if hoje == DiaDaSemana.Terca:
    print("Hoje é terça-feira.")
